airflow/models/taskinstance.py

```diff
@@ -539,7 +539,11 @@ def _refresh_from_db(
         task_instance.end_date = ti.end_date
         task_instance.duration = ti.duration
         task_instance.state = ti.state
-        task_instance.try_number = ti._try_number  # private attr to get value unaltered by accessor
+        try:
+            task_instance.try_number = ti._try_number  # private attr to get value unaltered by accessor
+        except AttributeError:
+            task_instance.try_number = ti.try_number  # the TaskInstancePydantic case
```

Comment:
Maybe we should make `_try_number` an alias (property) on TaskInstancePydantic instead? This seems too problematic.

Reply:
Yeah i'm actually working on that right now.  But I'm on a detour trying to get try number working properly from a logging perspective with db isolation enabled.  Try number immutability cannot come soon enough.

Reply:
ok @uranusjr I updated this PR to make pydantic try_number behavior more like orm model PTAL

Reply:
this might not be right, gonna mark draft again